ocaml/libs/tracing/tracing.ml

```diff
@@ -263,16 +297,25 @@ module Span = struct
 
   let get_context t = t.context
 
-  let start ?(attributes = Attributes.empty) ~name ~parent ~span_kind () =
-    let trace_id =
+  let start ?(attributes = Attributes.empty)
+      ?(trace_context : TraceContext.t option) ~name ~parent ~span_kind () =
+    let trace_id, extra_context =
       match parent with
       | None ->
-          Trace_id.make ()
+          (Trace_id.make (), TraceContext.empty)
       | Some span_parent ->
-          span_parent.context.trace_id
+          (span_parent.context.trace_id, span_parent.context.trace_context)
     in
     let span_id = Span_id.make () in
-    let context : SpanContext.t = {trace_id; span_id} in
+    let context : SpanContext.t =
+      {trace_id; span_id; trace_context= extra_context}
+    in
+    let context =
+      (* If trace_context is provided to the call, override any inherited trace context. *)
```

Comment:
Any "inherit trace context" comes from `~parent`. Is the idea that eventually `~parent` will disappear in favour of `trace_context`, or is the latter only used between service boundaries (e.g. when making a call to another process)?